amd/src/step_select.js

```diff
@@ -111,11 +111,21 @@ define(
                 // Then convert from its array-of-objects output format into a single object.
                 var curstep = $stepform.serializeArray().reduce(
                     function(finalobj, field) {
-
-                        // Filter out the sesskey and formslib system fields.
-                        if (field.name !== 'sesskey' && !field.name.startsWith('_qf__')) {
+                        // If field ends with [], the form el was an array.
+                        if (field.name.endsWith('[]')) {
+                            let fieldname = field.name.substring(0, field.name.length - 2);
+                            if (finalobj[fieldname] === undefined) {
+                                finalobj[fieldname] = [field.value];
```

Comment:
Should this just be ` finalobj[fieldname] = field.value;` instead of adding it in an array, as the else statement seems to just push the value without wrapping it.

Reply:
We want to preserve the array structure, rather than a single value. That way it makes it to JSON -> DB as an array

Reply:
Ah, just clicked as I was writing out an example. All good.